javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -62,19 +63,55 @@ predicate isConstant(Expr e) {
   isSymbolicConstant(e.(VarAccess).getVariable())
 }
 
+/**
+ * Holds if `e` directly uses a parameter's initial value as passed in from the caller.
+ */
+predicate isInitialParameterUse(Expr e) {
+  // unlike `SimpleParameter.getAnInitialUse` this will not include uses we have refinement information for
+  exists (SimpleParameter p, SsaExplicitDefinition ssa |
+    ssa.getDef() = p and
+    ssa.getVariable().getAUse() = e and
+    not p.isRestParameter()
+  )
+  or
+  isInitialParameterUse(e.(LogNotExpr).getOperand())
```

Comment:
The predicate qldoc doesn't obviously cover this case; could you extend it? Also, for consistency, could you add a similar case to the corresponding predicate in the heterogeneous-comparison query?

Reply:
There is no need to extend in `heterogeneous-comparison` case, the negation will coerce any value to a boolean value, regardless of inter-procedural information.

Reply:
Fair enough.